main/lsp/requests/rename.cc

```diff
@@ -44,12 +44,24 @@ bool isSubclass(const core::GlobalState &gs, core::SymbolRef root, core::SymbolR
     }
     visited[s.classOrModuleIndex()] = true;
 
-    auto super = s.data(gs)->superClass();
-    if (super.exists()) {
-        memoized[s.classOrModuleIndex()] = isSubclass(gs, root, super, memoized, visited);
+    if (s.data(gs)->derivesFrom(gs, root)) {
+        memoized[s.classOrModuleIndex()] = true;
     } else {
-        memoized[s.classOrModuleIndex()] = false;
+        auto super = s.data(gs)->superClass();
+        auto parents = s.data(gs)->mixins();
+        // Reverse parents so the last loaded mixin is first
+        reverse(parents.begin(), parents.end());
+        parents.emplace_back(super);
+        for (auto p : parents) {
+            if (p.exists()) {
+                auto result = isSubclass(gs, root, p, memoized, visited);
+                memoized[s.classOrModuleIndex()] = result;
```

Comment:
I don't understand this logic. It seems like the last parent encountered determines if rename derives from a given class, but that's not true?